docs/sapling.md

```diff
@@ -0,0 +1,277 @@
+
+const inMemorySpendingKey = await InMemorySpendingKey.fromMnemonic('YOUR_MNEMONIC');
+
+const saplingToolkit = new SaplingToolkit(
+    inMemorySpendingKey, 
+    { contractAddress: saplingContract.address, memoSize: 8 }, 
+    tezos.getFactory(RpcReadAdapter)()
+)
+```
+
+## How to retrieve my balance in the Sapling shielded pool?
+
+When calling the `getSaplingTransactionViewer` method of the `SaplingToolkit` class, an instance of the `SaplingTransactionViewer` class is returned. The `SaplingTransactionViewer` class allows retrieving and decrypting sapling transactions for the specified viewing key and calculating the unspent balance. 
+
+For each entry in the shielded pool, the `SaplingTransactionViewer` class will try to decrypt them using the viewing key as if it were the receiver. If a ciphertext is successfully decrypted, the configured account was the receiver of the output. The `SaplingTransactionViewer` will find which inputs were not spent by computing their nullifier. If an input is spent, its nullifier will be in the sapling state. If the nullifier is not present, thus the input has not been spent, and its value will be considered in the calculated balance.
+
+The balance can be retrieved as follow:
+
+```ts
+const txViewer = await saplingToolkit.getSaplingTransactionViewer();
+const initialBalance = await txViewer.getBalance();
+```
+
+Note that the balance is represented in mutez.
+
+## How to retrieve my transaction history?
+
+The `SaplingTransactionViewer` class exposes a method called `getIncomingAndOutgoingTransactions` which allows decrypting the transactions received and sent based on the viewing key. Information like the value in mutez, the memo, the payment address can be retrieved as well as if the input is spent for the incoming ones.
+
+Example:
+
+```ts
+const txViewer = await saplingToolkit.getSaplingTransactionViewer();
+const initialBalance = await txViewer.getIncomingAndOutgoingTransactions();
+
+/* initialBalance :
+{
+    incoming: [
+    {
+        value: new BigNumber(3000000),
+        memo: 'First Tx',
+        paymentAddress: alicePaymentAddress,
+        isSpent: true
+    },
+    {
+        value: new BigNumber(1000000),
+        memo: '',
+        paymentAddress: alicePaymentAddress,
+        isSpent: false
+    }
+    ],
+    outgoing: [
+    {
+        value: new BigNumber(2000000),
+        memo: 'A gift',
+        paymentAddress: bobPaymentAddress
+    },
+    {
+        value: new BigNumber(1000000),
+        memo: '',
+        paymentAddress: alicePaymentAddress
+    }
+    ]
+} */
+```
+
+## How to prepare a shielded transaction?
+
+A shielded transaction allows sending tokens from a Tezos account (tz1, tz2, tz3) to an address (zet). The `prepareShieldedTransaction` method of the `SaplingToolkit` takes an array of `ParametersSaplingTransaction`, making it possible to send tez to multiple addresses at once if needed. 
```

Comment:
should we call it a Sapling address?  ..  address (zet)

Reply:
Yes, it will be more clear